lib/codegen/DecisionParser.cpp

```diff
@@ -197,27 +197,32 @@ class DTPreprocessor {
 
   DecisionNode *makeIterator(yaml_node_t *node) {
     std::string name = to_string(vec(get(node, "collection")));
-    llvm::Type *type = getParamType(
-        KORECompositeSort::getCategory(str(get(node, "sort"))), mod);
+    std::string cat = str(get(node, "sort"));
+    llvm::Type *type = getParamType(KORECompositeSort::getCategory(cat), mod);
     std::string function = str(get(node, "function"));
     auto child = (*this)(get(node, "next"));
 
     return MakeIteratorNode::Create(
         name, type, name + "_iter",
-        llvm::PointerType::getUnqual(getTypeByName(mod, "iter")), function,
-        child);
+        llvm::PointerType::get(
+            getTypeByName(mod, cat == "SET.Set" ? "setiter" : "mapiter"), 1),
```

Comment:
Since you added similar assertions elsewhere, maybe add one here too for uniformity.